pyproject.toml

```diff
@@ -43,6 +43,7 @@ dependencies = [
     "anytree>=2.8.0",
     "sympy>=1.12",
     "typing-extensions>=4.10.0",
+    "pandas>=1.5.0"
```

Comment:
I do not think this change is desirable. Was there a reason to include `pandas` as a required dependency?

Reply:
The solution class needs pandas, so it probably is a required dependency:
![image](https://github.com/pybamm-team/PyBaMM/assets/10170302/729eab6d-5081-479b-b40e-721af5d0a2c0)